src/main/java/seedu/address/model/lesson/Date.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.lesson;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Lesson's date in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class Date {
+
+    public static final String MESSAGE_CONSTRAINTS = "Dates should be of the format dd MMM yyyy "
+            + "and adhere to the following constraints:\n"
+            + "1. dd and yyyy are numerical characters.\n"
+            + "2. MMM are alphabetical characters. e.g. Jan, Feb, ..., Dec";
+
+
+    public static final String VALIDATION_REGEX = "^[0-9]{2}\\s[a-zA-Z]{3}\\s[0-9]{4}";
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy");
+
+    public final String value;
+
+    /**
+     * Constructs an {@code Date}.
+     *
+     * @param date A valid Date.
+     */
+    public Date(String date) {
+        requireNonNull(date);
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
+        value = date;
+    }
+
+    /**
+     * Returns if a given string is a valid date.
+     */
+    public static boolean isValidDate(String test) {
+        boolean isValid = true;
+        try {
+            LocalDate.parse(test, FORMATTER);
+        } catch (DateTimeParseException e) {
+            isValid = false;
+        } finally {
+            return test.matches(VALIDATION_REGEX) && isValid;
+        }
+    }
+
```

Comment:
Might need a custom, stricter check for dates. For example, 31 Feb should not be valid.

Reply:
I think the `LocalDate.parse(test, FORMATTER)` will take care of the invalid dates and throw DateTimeParseException, so there is not explicit need for a custom check for dates.